lua/neogit.lua

```diff
@@ -111,6 +111,42 @@ function M.open(opts)
   end
 end
 
+-- EXPERIMENTAL!
+-- This can be used to create bindable functions for custom keybindings:
+--   local neogit = require("neogit")
+--   vim.keymap.set('n', '<leader>gcc', neogit.action('commit', 'commit', { '--verbose', '--all' }))
+--
+---@param popup  string Name of popup, as found in `lua/neogit/popups/*`
+---@param action string Name of action for popup, found in `lua/neogit/popups/*/actions.lua`
+---@param args   table? CLI arguments to pass to git command
+---@return function
+function M.action(popup, action, args)
+  local notification = require("neogit.lib.notification")
+  local a = require("plenary.async")
+
+  return function()
+    a.run(function()
+      local ok, actions = pcall(require, "neogit.popups." .. popup .. ".actions")
```

Comment:
How will this handle populating the `env` and `popup` field of the actions first parameter?

Will the user need to supply a dummy `Popup` instance?

Reply:
The `env` gets skipped entirely: this bypasses the popup, and all associated state, to allow easy access to the _actions_.

They user can pass a table of CLI flags that populate the return value of a mock `get_arguments()` function. This _won't_ work for stuff that we use internally like `--graph`, `--color`, and `--decorate` for `git log`, but thats it.

The value of exposing an API like this is that a user could pre-compose a set of CLI arguments and bind that to a key. Then, they still get all the advantages of neogit like the `client.wrap` RPC stuff, more custom function like `spin_out`, etc. 

_I_ still think the UI is the way to go, but I don't see the harm in this. The biggest downside of exposing this is that we're making the popup names/action names part of the public api, and changing them becomes a breaking change. Which... I think thats fine. Don't see why we would rename them.